### DISCUSSION TITLE
Can't get `navbar-start | navbar-center | navbar-end` to work.

### DISCUSSION THREAD

**Discussion by abohannon** [saadeghi/daisyui#1402] [Q&A]
I'm working on a navbar and none of the positioning classes render any styles. Anyone else experiencing this issue? I thought it might have been clashing styles from another styling solution in our repo, but I spun up a fresh NextJS + Tailwind repo, added daisyui, and no luck.

<img width="636" alt="Screen Shot 2022-11-29 at 7 34 49 PM" src="https://user-images.githubusercontent.com/889536/204702118-c51accb3-cb1d-4442-b711-8a18e62314d7.png">

**Comment by saadeghi** [comment]
Please make sure the `content` config on your `tailwind.config.js` file includes path to all your markup files.

If that's not the issue, please provide a sample repo on GitHub so I can reproduce the issue.

**Reply by abohannon** [reply]
I actually figured out the issue. We were dynamically constructing the navbar classes and Tailwind can't recognize them when it creates them. Here's a Stack Overflow issue discussing it with more documentation: https://stackoverflow.com/a/71819163

You need to pass the full classname intact and do any dynamic evaluation like this (we use the "classnames" library for this sort of thing):

```
const classes = classNames({
 [`navbar-${section}`]: section                  // doesn't work
 'navbar-start': section === 'start',            // does work
 'navbar-center': section === 'center',
 'navbar-end': section === 'end',
})

return <div classNames={classes>{children}</div>
```

The daisy-ui react library uses the first approach for a similar component and they use `twMerge` but I tried that and it didn't work for us. Anyway, hopefully this is helpful for anyone in the future who runs into this.
